multihash_iterator.go

```diff
@@ -66,3 +70,33 @@ func (it *sliceMhIterator) Next() (multihash.Multihash, error) {
 	it.pos++
 	return mh, nil
 }
+
+var _ MultihashIterator = (*ipldMapMhIter)(nil)
```

Comment:
nit: Not necessary, because `HamtMultihashIterator` returns `&ipldMapMhIter` as `MultihashIterator` thereby guaranteeing that the interface is implemented.

Reply:
This is more as a documentation that the structs should implement `MultihashIterator`